engine/apps/slack/models/slack_message.py

```diff
@@ -18,24 +18,48 @@
 
 if typing.TYPE_CHECKING:
     from apps.alerts.models import AlertGroup
+    from apps.base.models import UserNotificationPolicy
+    from apps.slack.models import SlackChannel
+    from apps.user_management.models import User
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
 
 class SlackMessage(models.Model):
     alert_group: typing.Optional["AlertGroup"]
+    channel: "SlackChannel"
 
     id = models.CharField(primary_key=True, default=uuid.uuid4, editable=False, max_length=36)
-
     slack_id = models.CharField(max_length=100)
 
-    # TODO: convert this to a foreign key field to SlackChannel
-    channel_id = models.CharField(max_length=100, null=True, default=None)
+    _channel_id = models.CharField(max_length=100, null=True, default=None)
+    """
+    DEPRECATED/TODO: this is no longer being referenced/set, drop in a separate PR/release
+    """
+
+    channel = models.ForeignKey(
+        "slack.SlackChannel", on_delete=models.CASCADE, null=True, default=None, related_name="slack_messages"
+    )
+    """
+    TODO: once we've migrated the data in `_channel_id` to this field, set `null=False`
+    as we should always have a `channel` associated with a message
+    """
 
     organization = models.ForeignKey(
-        "user_management.Organization", on_delete=models.CASCADE, null=True, default=None, related_name="slack_message"
+        "user_management.Organization",
+        on_delete=models.CASCADE,
+        null=True,
+        default=None,
+        related_name="slack_message",
     )
+    """
+    TODO: do we really need organization here? We now have a foreign key reference to channel, which has a foreign
+    key reference to slack_team_identity, which has a foreign key reference to organization...
```

Comment:
Hmm right, good call out. I think I slightly misunderstood the relationship, thanks for clarifying 👍
- I assumed that every `SlackChannel` was unique for a given stack's (org) workspace installation.. ex. if you have stack/org A, and stack/org B both installed to workspace X (which has 1 slack channel, Y), you'd have two `slack_team_identity` objects and two `slack_channel` objects, when in reality its only 1 `slack_team_identity` and 1 `slack_channel`

To me it feels duplicative to have a relationship to _both_ `organization` **and** `slack_team_identity` on this table. Given what you said (and my, hopefully correct, understanding) maybe what we could do here instead is, when creating, just set `SlackMessage.organization`, and when we need to reference the `slack_team_identity`, you'd be able to just do `slack_message.organization.slack_team_identity`. This would allow us to (eventually) drop the relationship to `slack_team_identity` on this table. wdyt?

Reply:
yes I think this makes sense 👍 not sure how big of a refactor this would be, but I agree with the idea that having both `organization` and `slack_team_identity` on `SlackMessage` is not optimal